### PR TITLE
Add an error for unhandled misaligned record member

### PR DESCRIPTION
**Pull request by daviditen**
For now, instead of adding padding for llvm struct members that are not aligned
issue an error saying that misaligned members are not currently handled.

Update the .bad file and prediff in a .future test to expect the error.

Signed-off-by: David Iten <daviditen@users.noreply.github.com>

### CODE REVIEW

**Comment by daviditen** [comment]
Passed Linux64 paratest with `-compopts --llvm`.

File: compiler/codegen/type.cpp [hunk 1]
```diff
@@ -423,10 +430,25 @@ void AggregateType::codegenDef() {
...
 
-
         llvm::StructType* stype = llvm::cast<llvm::StructType>(type);
         stype->setBody(params);
 
+        if (aligns.size() == params.size()) {
+          for (int i = 0; i < params.size(); i++) {
+            unsigned offset = info->module->getDataLayout().getStructLayout(stype)->getElementOffset(i);
+#if HAVE_LLVM_VER >= 100
+            unsigned align = 1 << Log2(aligns[i].valueOrOne());
```
Comment (mppf) [inline compiler/codegen/type.cpp:440]: What's the point of `1 << Log2(x)` ? Isn't it the same as `unsigned align = aligns[i].valueOrOne()` ?

File: test/llvm/alignment/check-align-padding.bad [hunk 2]
```diff
@@ -1,7 +1 @@
-%int32int64_chpl = type { i32, i64 }
```
Comment (mppf) [inline test/llvm/alignment/check-align-padding.bad:1]: Can you split this test into 2 cases - one where it compiles and we the alignment is OK at runtime - and another where we see this error?

**Review by mppf — APPROVED** [review]

File: compiler/codegen/type.cpp (hunk 1 above)
Comment (daviditen) [reply compiler/codegen/type.cpp:440]: `MaybeAlign.valueOrOne()` returns an `Align` class. `Align` has a `value` method, but its documentation says "This is a hole in the type system and should not be abused." so I used `Log2` instead to get the alignment out as an unsigned int.

File: test/llvm/alignment/check-align-padding.bad (hunk 2 above)
Comment (daviditen) [reply test/llvm/alignment/check-align-padding.bad:1]: I split it into two tests as you requested.

Comment (mppf) [reply test/llvm/alignment/check-align-padding.bad:1]: Shouldn't this .bad file go away?

File: test/llvm/alignment/check-align-padding.chpl [hunk 3]
```diff
@@ -38,8 +43,18 @@ record int64aligned16b {
   var field2: aligned16test;
 }
 
-proc compute(arg) {
+proc compute(ref arg) {
```
Comment (mppf) [inline test/llvm/alignment/check-align-padding.chpl:46]: Let's add something like `writeln("working with ", arg.type:string)` here so we can understand the test output better.

File: test/llvm/alignment/check-align-padding.bad (hunk 2 above)
Comment (daviditen) [reply test/llvm/alignment/check-align-padding.bad:1]: The .bad file matches the current behavior of the .future, so I think it should stay.

File: test/llvm/alignment/check-align-padding.chpl (hunk 3 above)
Comment (daviditen) [reply test/llvm/alignment/check-align-padding.chpl:46]: OK, done.

**Review by mppf — APPROVED** [review]

**Pull request merged**